### PR TITLE
Add integration tests for the payment reconciler lambda

### PR DESCRIPTION
**Pull request by BSaunders95**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,5 +7,7 @@ payment-reconciler*
 # Lint files
 lint.txt
 
+.DS_Store
 
-.DS_Store 
+# Test coverage file
+coverage.out
```
Comment (emead) [inline .gitignore]: This isn't specified in the styleguides. Should the styleguides be updated?

File: Makefile [hunk 2]
```diff
@@ -26,6 +26,10 @@ test: test-unit
 test-unit:
 	go test $(TESTS) -run 'Unit' -coverprofile=coverage.out
 
+.PHONY: test-integration
+test-integration:
+	go test $(TESTS) -run 'Integration' -coverprofile=coverage.out
```
Comment (emead) [inline Makefile]: This doesn't correspond with the styleguides

File: README.md [hunk 3]
```diff
@@ -26,3 +26,18 @@ SFTP_PORT                                        | The port over which to connec
 SFTP_USERNAME                                    | The username of the SFTP server credentials.                                                                  | 
 SFTP_PASSWORD                                    | The password of the SFTP server credentials.                                                                  |
 SFTP_FILE_PATH                                   | The file path, relative to the root of the SFTP server, to which to upload CSV files.                         | 'uploadPath' (will result is CV's uploaded to directory: ~/uploadPath)
+
+### Test Environment Variables
+Environment variables required to execute integration tests for this lambda:
+
+Name                | Description                                                                           | Examples
+--------------------| --------------------------------------------------------------------------------------|--------------------------------------------------------------------------
+ENV_REGION_AWS      | The AWS region code for the lambda.                                                   | 'eu-west-2'
```
Comment (emead) [inline README.md]: Backticks would give better formatting for the examples
```suggestion
ENV_REGION_AWS      | The AWS region code for the lambda.                                                   | `eu-west-2`
```

File: config/config.go [hunk 4]
```diff
@@ -10,11 +10,16 @@ type Config struct {
...
-	SFTPPort               string `env:"SFTP_PORT"                                    flag:"sftp-port"                                     flagDesc:"Port to connect to the SFTP server"`
-	SFTPUserName           string `env:"SFTP_USERNAME"                                flag:"sftp-username"                                 flagDesc:"Username of SFTP server"`
-	SFTPPassword           string `env:"SFTP_PASSWORD"                                flag:"sftp-password"                                 flagDesc:"Password of SFTP server"`
-	SFTPFilePath           string `env:"SFTP_FILE_PATH"                               flag:"sftp-file-path"                                flagDesc:"File path on the SFTP server"`
+	SFTPConfig
+}
+
+// SFTPConfig holds configuration details specific to an SFTP connection.
+type SFTPConfig struct {
+	SFTPServer   string `env:"SFTP_SERVER"                                  flag:"sftp-server"                                   flagDesc:"Name of the SFTP server"`
```
Comment (emead) [inline config/config.go]: Lots of unnecessary whitespace on these lines

File: lambda/lambda_test.go [hunk 5]
```diff
@@ -154,3 +163,112 @@ func TestUnitExecute(t *testing.T) {
...
+
+	cfg, err := config.GetTestConfig()
+	if err != nil {
+		t.Errorf("Failed to fetch test config: %s", err)
+		t.FailNow()
+	}
+
+	if cfg.NotPopulated() {
+		t.Log("Some config required for integration tests is not populated. Skipping test")
+		t.SkipNow()
```
Comment (emead) [inline lambda/lambda_test.go:180]: Would it be better to fail, rather than skip the tests?
If the tests are skipped, they are likely to be ignored.

Comment (BSaunders95) [reply lambda/lambda_test.go:180]: If at some point someone decided to add the `test-integration` target to the `test` target, I didn't want to tests to fail should someone not have the required env variables set, so I figured it best to log a message and bail out. Happy to change if you have a massive preference though? 👍

File: config/config.go (hunk 4 above)
Comment (BSaunders95) [reply config/config.go]: Removed 👍

File: README.md (hunk 3 above)
Comment (BSaunders95) [reply README.md]: Updated all examples to use backticks

File: Makefile (hunk 2 above)
Comment (BSaunders95) [reply Makefile]: `coverprofile` flag removed 👍

File: .gitignore (hunk 1 above)
Comment (BSaunders95) [reply .gitignore]: Removed as this file was produced from a flag which has also been removed

**Review by emead — APPROVED** [review]
